NumberBaseball/Source.c

```diff
@@ -0,0 +1,150 @@
+#include <stdio.h>
+#include <stdlib.h> 
+#include<time.h> 
+
+#define LINESIZE 1024
+#define NUMBER_OF_GUESSES 3
+#define STRIKES_TO_END 3
+
+int numStrikes;
+int numBalls;
+int numOuts;
+int resultStrikes;
+int resultBalls;
+int resultOuts;
+int n1;
+int n2;
+int n3;
+int random1;
+int random2;
+int random3;
+
+void Init() {
+	numStrikes = 0;
+	numBalls = 0;
+	numOuts = 0;
+	resultStrikes = 0;
+	resultBalls = 0;
+	resultOuts = 0;
+	printf("Init\n");
+}
+
+void Update() {
+	char line[LINESIZE];
+
+	resultStrikes = 0;
+	resultBalls = 0;
+	resultOuts = 0;
+
+	while (1) {
+		printf("Please enter your %d numbers:\n", NUMBER_OF_GUESSES);
+		if (!fgets(line, LINESIZE, stdin)) /* if we fail to read a line */
+		{
+			clearerr(stdin);
+			break;
+		}
+		if (sscanf_s(line, "%d %d %d", &n1, &n2, &n3) == NUMBER_OF_GUESSES && (n1 != n2 && n1 != n3 && n2 != n3))
+		{
+			/* Use current time as seed for random generator */
+			srand(time(0));
+			/* Generates random number between 0 and 9 */
+			random1 = rand() % 10;
+			do
```

Comment:
This code below is too long.
you can separate it by making other functions for these, and it's good for maintainability